toolchain/check/handle_impl.cpp

```diff
@@ -471,14 +476,14 @@ static auto BuildImplDecl(Context& context, Parse::AnyImplDeclId node_id,
         context.emitter().Emit(loc, ImplUnusedBinding);
         // Don't try to match the impl at all, save us work and possible future
         // diagnostics.
-        FillImplWitnessWithErrors(context, impl_info);
-        context.impls().Get(impl_decl.impl_id).witness_id =
-            SemIR::ErrorInst::InstId;
+        FillImplWitnessWithErrors(context, stored_impl_info);
+        stored_impl_info.witness_id = SemIR::ErrorInst::InstId;
       }
     }
   } else {
-    auto prev_decl_generic_id =
-        context.impls().Get(impl_decl.impl_id).generic_id;
+    auto& stored_impl_info = context.impls().Get(impl_decl.impl_id);
+
+    auto prev_decl_generic_id = stored_impl_info.generic_id;
     FinishGenericRedecl(context, prev_decl_generic_id);
```

Comment:
Maybe use the `stored_impl_info` directly?
```suggestion
    FinishGenericRedecl(context, stored_impl_info.generic_id);
````

Reply:
Done